### PR TITLE
sokol-flex: clean up unnecessary items

### PR DESCRIPTION
**Pull request by kergoth**
Clean up various unnecessary items from our DISTRO.

JIRA: SB-20618

### CODE REVIEW

**Comment by kergoth** [comment]
This still has two open questions:

- `do_image_complete[recrdeptask] += "do_populate_lic"` is this still needed?
- Is `IMAGE_ROOTFS_EXTRA_SPACE` necessary when most builds use SD/MMC boot which resizes on first boot? It's also set to 0 in bitbake.conf and many other images.

I'm looking into the first of these now. The latter requires discussion.

**Comment by abelal** [comment]
I think we're ok to drop IMAGE_ROOTFS_EXTRA_SPACE

File: meta-sokol-flex-distro/conf/distro/sokol-flex.conf [hunk 1]
```diff
@@ -192,10 +192,6 @@ INHERIT += "buildhistory"
 # Copy the image license manifest into DEPLOY_DIR_IMAGE
 INHERIT += "deploy-license-manifest"
 
-# Warn if there are available update layers which are not included in the
-# current configuration
-INHERIT += "flex-updates-check"
```
Comment (abelal) [inline meta-sokol-flex-distro/conf/distro/sokol-flex.conf:207]: Aren't we checking the inclusion of meta-flex-cve using this class?

File: meta-sokol-flex-common/classes/flex-updates-check.bbclass [hunk 2]
```diff
@@ -1,34 +0,0 @@
-# ---------------------------------------------------------------------------------------------------------------------
```
Comment (abelal) [inline meta-sokol-flex-common/classes/flex-updates-check.bbclass:1]: relates to the comment on the previous commit.

**Comment by abelal** [comment]
Apart from one observation against flex-updates-check this is good to go.

**Comment by ahsanhussain** [comment]
Can we also remove UBI_VOLNAME, APPEND and friends because we take care of such things differently in the BSP layer.

File: meta-sokol-flex-distro/conf/distro/sokol-flex.conf (hunk 1 above)
Comment (kergoth) [reply meta-sokol-flex-distro/conf/distro/sokol-flex.conf:207]: Erk, yes, thanks, well spotted :)

**Comment by kergoth** [comment]
[**@ahsanhussain**](https://github.com/ahsanhussain) commented on [Nov 1, 2022, 7:17 AM MST](https://github.com/MentorEmbedded/meta-sokol-flex/pull/70#issuecomment-1298578310 "2022-11-01T14:17:30Z - Replied by Github Reply Comments"):
> Can we also remove UBI_VOLNAME, APPEND and friends because we take care of such things differently in the BSP layer.

The default UBI_VOLNAME includes the machine name, which we felt was redundant, it's not needed in filenames as DEPLOY_DIR_IMAGE includes MACHINE, and it's not needed on target, where we already know what we're booting, so this seems like a general fix more than machine-specifics.

The APPEND bits set in the distro are the bits of kernel configuration which aren't solely the realm of the BSP, such as splash. While we aren't obeying it in our bsps currently, I'd rather see us start doing so than scratch it entirely.

File: meta-sokol-flex-distro/conf/distro/sokol-flex.conf (hunk 1 above)
Comment (kergoth) [reply meta-sokol-flex-distro/conf/distro/sokol-flex.conf]: Fixed

**Review by abelal — APPROVED** [review]

**Comment by ahsanhussain** [comment]
We're already doing some of the bootargs related stuff corresponding to our features in flex-distro.h implementation. @abelal can we used APPEND bits in parallel, or does it make sense to use only one of these?

**Comment by abelal** [comment]
> We're already doing some of the bootargs related stuff corresponding to our features in flex-distro.h implementation but that's only for U-Boot. @abelal can we used APPEND bits in parallel? Should we do it in the BSP layer like flex-distro.h?

AFAIK the APPEND syntax is only applicable to grub-efi based configurations so @ahsanhussain has a point.

**Review by ahsanhussain — APPROVED** [review]
After some discussion with @abelal I've created a separate improvements task for APPEND mechanism: https://jira.alm.mentorg.com/browse/SB-20844

This is good to merge.

**Pull request merged**